packages/next/build/index.ts

```diff
@@ -614,49 +611,37 @@ export default async function build(
     await (async () => {
       // IIFE to isolate locals and avoid retaining memory too long
       const runWebpackSpan = nextBuildSpan.traceChild('run-webpack-compiler')
+
+      const commonWebpackOptions = {
+        buildId,
+        config,
+        hasReactRoot,
+        pagesDir,
+        reactProductionProfiling,
+        rewrites,
+        runWebpackSpan,
+        target,
+      }
+
       const configs = await runWebpackSpan
         .traceChild('generate-webpack-config')
         .traceAsyncFn(() =>
           Promise.all([
             getBaseWebpackConfig(dir, {
-              buildId,
-              reactProductionProfiling,
-              isServer: false,
-              config,
-              target,
-              pagesDir,
+              ...commonWebpackOptions,
+              compilerType: 'client',
               entrypoints: entrypoints.client,
-              rewrites,
-              runWebpackSpan,
-              hasReactRoot,
             }),
             getBaseWebpackConfig(dir, {
-              buildId,
-              reactProductionProfiling,
-              isServer: true,
-              config,
-              target,
-              pagesDir,
+              ...commonWebpackOptions,
+              compilerType: 'server',
               entrypoints: entrypoints.server,
-              rewrites,
-              runWebpackSpan,
-              hasReactRoot,
             }),
-            hasReactRoot
-              ? getBaseWebpackConfig(dir, {
-                  buildId,
-                  reactProductionProfiling,
-                  isServer: true,
-                  isEdgeRuntime: true,
-                  config,
-                  target,
-                  pagesDir,
-                  entrypoints: entrypoints.edgeServer,
-                  rewrites,
-                  runWebpackSpan,
-                  hasReactRoot,
-                })
-              : null,
+            getBaseWebpackConfig(dir, {
```

Comment:
Do we want to always call this even if `hasReactRoot` is `false`? Users that have the custom webpack function in their `next.config.js` may be confused when it's called three times if they aren't already using react 18.